linkerd/app/outbound/src/opaq/logical.rs

```diff
@@ -1,43 +1,66 @@
-use super::{Concrete, Logical};
+use super::concrete;
 use crate::Outbound;
 use linkerd_app_core::{
     io,
     profiles::{self, Profile},
-    proxy::api_resolve::ConcreteAddr,
-    svc, Error,
+    proxy::{api_resolve::Metadata, tcp::balance},
+    svc,
+    transport::addrs::*,
+    Error, Infallible, NameAddr,
 };
 use linkerd_distribute as distribute;
+use std::{fmt::Debug, hash::Hash, time};
+use tokio::sync::watch;
 
-#[cfg(tests)]
+#[cfg(test)]
 mod tests;
 
+#[derive(Clone, Debug)]
+pub enum Logical {
+    Route(NameAddr, profiles::Receiver),
+    Forward(Remote<ServerAddr>, Metadata),
+}
```

Comment:
naming bikeshed (significance: silly, feel free to ignore): are both variants of this actually "logical" addresses? the `Forward` case represents a concrete address, and it's really only the `Route` case that's a logical address, so it's really the `Routable` target type that's a logical target. but...i dunno what we would call this type instead... :woman_shrugging:

Reply:
I had initially called it `Target`, but that seemed too overloaded. I think the
name will change as we introduce real config types...